lib/spack/spack/build_systems/cmake.py

```diff
@@ -458,19 +458,22 @@ def cmake(
     ) -> None:
         """Runs ``cmake`` in the build directory"""
 
-        # skip cmake phase if it is an incremental develop build
-        # These are the files that will re-run CMake that are generated from a successful
-        # configure step
-        primary_generator = _extract_primary_generator(self.generator)
-        if primary_generator == "Unix Makefiles":
-            configure_artifact = "Makefile"
-        elif primary_generator == "Ninja":
-            configure_artifact = "ninja.build"
-
-        if spec.is_develop and os.path.isfile(
-            os.path.join(self.build_directory, configure_artifact)
-        ):
-            return
+        if spec.is_develop:
+            # skip cmake phase if it is an incremental develop build
+
+            def _configure_artifact():
```

Comment:
Why is this inline function added? It's called once, it doesn't capture anything. It's just more code.

The problem I pointed to isn't fixed. The function returns Optional[str] not str. Can you just do

```python
configure_artifact = "Makefile"

if <something>:
    configure_artifact = "ninja.build"
```

Reply:
It'd be nice if you're going to take the time to type out the exact code you want if you'd just put it as a suggestion. It would save us both some time. I can fix the return type. That was not mentioned in your previous comments.